tutorials/plot_background_statistics.py

```diff
@@ -0,0 +1,496 @@
+#   independently, but still has a broadened estimate.
+# - **Clustering with TFCE and "hat"** tightens the area declared
+#   significant (again FWER corrected).
+#
+# Statistical functions in MNE
+# ----------------------------
+# MNE provides several convenience parametric testing functions that can be
+# used in conjunction with the non-parametric clustering methods. However,
+# the set of functions we provide is not meant to be exhaustive.
+#
+# If the univariate statistical contrast of interest to you is not listed
+# here (e.g., interaction term in an unbalanced ANOVA), consider checking
+# out the :mod:`statsmodels` package. It offers many functions for computing
+# statistical contrasts, e.g., :func:`statsmodels.stats.anova.anova_lm`.
+# To use these functions in clustering:
+#
+# 1. Determine which statistical test you would use in a univariate context
+#    (e.g., to compute your contrast of interest if there only a single
+#    output, like reaction times).
+# 2. Wrap the call to that function within a function that takes an input of
+#    the same shape that is expected by your clustering function,
+#    and returns an array of the same shape without the "samples" dimension
+#    (e.g., :func:`mne.stats.permutation_cluster_1samp_test` takes an array
+#    of shape ``(n_samples, p, q)`` and returns an array of shape ``(p, q)``).
+# 3. Pass this wrapped function to the ``stat_fun`` argument to the clustering
+#    function.
+# 4. Set an appropriate ``threshold`` value (float or dict) based on the
+#    values your statistical contrast function returns.
+#
+# Parametric methods provided by MNE
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# - :func:`mne.stats.ttest_1samp_no_p`
+#     Paired t test, optionally with hat adjustment.
+#     This is used by default for contrast enhancement in paired cluster tests.
+#
+# - :func:`mne.stats.f_oneway`
+#     One-way ANOVA for independent samples.
+#     This can be used to compute various F-contrasts. It is used by default
+#     for contrast enhancement in non-paired cluster tests.
+#
+# - :func:`mne.stats.f_mway_rm`
+#     M-way ANOVA for repeated measures and balanced designs.
+#     This returns F-statistics and p-valus. The associated helper function
+#     :func:`mne.stats.f_threshold_mway_rm` can be used to determine the
+#     F-threshold at a given significance level and set of degrees of freedom.
+#
+# - :func:`mne.stats.linear_regression`
+#     Compute ordinary least square regressions on multiple targets, e.g.,
+#     sensors, time points across trials (samples).
+#     For each regressor it returns the beta values, t-staistics, and
+#     uncorrected significance values. While it can be used as a test it is
+#     particularly useful to compute weighted averages.
+#
+# Non-parametric methods
+# ^^^^^^^^^^^^^^^^^^^^^^
+#
+# - :func:`mne.stats.permutation_cluster_test`
+#     Unpaired contrasts with spatial connectivity.
```

Comment:
why only spatial connectivity? isn't this the general function that doesn't assume in which dimension the connectivity structure is applied?